src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -50,6 +51,29 @@
 public class JavadocDetailNodeParser {
 
     /**
+     * Options for errors listeners attached to ANTLR parser.
+     */
+    public enum ErrorListenerOption {
```

Comment:
Do you mean we should suppress it ? `enum` looks much clearer to me. Plus the user also gets a proper javadoc for `enum`. Plus ENUM gives us the opportunity to have even more errorlistener options in future.